docs/PORTING.md

```diff
@@ -119,11 +119,11 @@ int     flash_area_erase(const struct flash_area *, uint32_t off, uint32_t len);
 uint8_t flash_area_align(const struct flash_area *);
 /*< What is value is read from erased flash bytes. */
 uint8_t flash_area_erased_val(const struct flash_area *);
-/*< Reads len bytes from off, and checks if the read data is erased. Returns
-    1 if empty (that is containing erased value), 0 if not-empty, and -1 on
-    failure. */
-int     flash_area_read_is_empty(const struct flash_area *fa, uint32_t off,
-                     void *dst, uint32_t len);
+/*< Verify that the data in buffer is erased according to the erase value of
+    the given flash area device. Returns true if the buffer is erased; false
+    if it is not erased, or buffer==NULL, or len==0. */
+bool    flash_area_buffer_is_erased(const struct flash_area *fa,
```

Comment:
Why does every back-end have to implement this?  Aren't they all the same?  Could it be put into common code somewhere?  `buffer_is_erased(data, size, erase_value)`?

Reply:
Done, thanks for the hint!

Reply:
It isn't always the same. For example, the LPC55S69 has a command in the flash controller that is able to do this read check with different voltage thresholds to be able to check truly erased vs written as erased.

Reply:
> to be able to check truly erased vs written as erased.

As long as it is erased, "truly" or "written", shouldn't it be the same from our perspective?